### PR TITLE
Refactor task_counts and add incomplete_tasks

### PR DESCRIPTION
**Pull request by rashidg**
Refs #SDE-4241

TODO:
- [x] Set milestone label in pull request (SDE version)
- [x] Updated changelog.md
- [x] Completed documentation for the new endpoint.
- [x] Feature has been merged into SDE develop.

### CODE REVIEW

File: source/includes/changelog.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 
 ## v4.6
+- Add incomplete_tasks include filter and redesign task_counts include filter on projects endpoint.
```
Comment (funkaoshi) [inline source/includes/changelog.md]: This is a v4.7  change? You duplicated the v4.6 header. Also why isn't this merged yet if it's in v4.7 or v4.6?

Comment (rashidg) [inline source/includes/changelog.md]: It's not merged yet, I will merge it to develop today. I think I was going to update the version but I forgot to ask. What version is develop?

**Comment by RouganStriker** [comment]
Can we merge this?

**Comment by rashidg** [comment]
Thanks @RouganStriker, I didn't get a +1 so I forgot to merge this

**Pull request merged**